src/Georadix.Core.Tests/GenericEqualityComparerFixture.cs

```diff
@@ -0,0 +1,214 @@
+                        new Model() { SubModelProp = new SubModel() { DateTimeProp = DateTime.Now } },
+                        new Model() { SubModelProp = new SubModel() },
+                        false
+                    }
+                };
+            }
+        }
+
+        public static IEnumerable<object[]> EqualsWithSubModelsScenarios
+        {
+            get
+            {
+                return new object[][]
+                {
+                    new object[]
+                    {
+                        sampleSubModel,
+                        sampleSubModel,
+                        true
+                    },
+                    new object[]
+                    {
+                        new SubModel(),
+                        new SubModel(),
+                        true
+                    },
+                    new object[]
+                    {
+                        new SubModel(new DateTime(9876543210), Guid.Empty, 999, "SubModel string"),
+                        new SubModel(new DateTime(9876543210), Guid.Empty, 999, "SubModel string"),
+                        true
+                    },
+                    new object[]
+                    {
+                        new SubModel(),
+                        null,
+                        false
+                    },
+                    new object[]
+                    {
+                        new SubModel(),
+                        new SubModel() { DateTimeProp = DateTime.Now },
+                        false
+                    },
+                    new object[]
+                    {
+                        new SubModel() { IntProp = 10 },
+                        new SubModel() { IntProp = -10 },
+                        false
+                    }
+                };
+            }
+        }
+
+        [Theory]
+        [PropertyData("EqualsWithModelsScenarios")]
+        public void EqualsWithModelsReturnsExpectedResult(Model x, Model y, bool expected)
+        {
+            var sut = new GenericEqualityComparer<Model>();
+
+            Assert.Equal(expected, sut.Equals(x, y));
+        }
+
+        [Theory]
+        [PropertyData("EqualsWithSubModelsScenarios")]
+        public void EqualsWithSubModelsReturnsExpectedResult(SubModel x, SubModel y, bool expected)
+        {
+            var sut = new GenericEqualityComparer<SubModel>();
+
+            Assert.Equal(expected, sut.Equals(x, y));
+        }
+
+        [Fact]
+        public void GetHashCodeForNullObjectReturnsZero()
+        {
+            var sut = new GenericEqualityComparer<Model>();
+
+            Assert.Equal(0, sut.GetHashCode(null));
+        }
+
+        [Fact]
+        public void GetHashCodeForValidObjectReturnsObjectHashCode()
+        {
+            var sut = new GenericEqualityComparer<Model>();
+            var model = new Model();
+
+            Assert.Equal(model.GetHashCode(), sut.GetHashCode(model));
+        }
+
+        public class Model
```

Comment:
Good catch. Enums worked fine as is, but changes were needed to support enumerables. Please review the latest changes since it was a bit involved on the reflection side.